libs/components/indicators/testing/src/label/label-harness.ts

```diff
@@ -0,0 +1,64 @@
+import { HarnessPredicate } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+import { SkyLabelType } from '@skyux/indicators';
+
+import { SkyLabelHarnessFilters } from './label-harness-filters';
+
+/**
+ * Harness for interacting with a label component in tests.
+ * @internal
+ */
+export class SkyLabelHarness extends SkyComponentHarness {
+  public static hostSelector = 'sky-label';
+
+  #getWrapper = this.locatorFor('.sky-label');
+  #getTextEl = this.locatorFor('.sky-label-text');
+  #getScreenReaderTextEl = this.locatorFor('.sky-screen-reader-only');
+
+  // #getAccessibilityEl = this.locatorForAll
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyLookupHarness` that meets certain criteria.
+   */
+  public static with(
+    filters: SkyLabelHarnessFilters
+  ): HarnessPredicate<SkyLabelHarness> {
+    return SkyLabelHarness.getDataSkyIdPredicate(filters);
+  }
+
+  /**
+   * Gets the text of the label component.
+   */
+  public async getLabelText(): Promise<string> {
+    return await (await this.#getTextEl()).text();
+  }
+
+  /**
+   * Gets the screen reader text read in place of the icon of the label component.
+   */
+  public async getScreenReaderText(): Promise<string> {
```

Comment:
I'd break this out into two methods: `getDescriptionType()` and `getCustomDescription()`. `getDescriptionType()` returns the value specified on the component's `descriptionType` input, and `getCustomDescription()` returns the value of the `customDescription` input. I'd prefer not to return the text for preset description types as those are subject to change and could cause tests to break if consumers are testing against the specific text. I'd also consider throwing an error when calling `getCustomDescription()` when `descriptionType` is not `custom`.